vault/expiration.go

```diff
@@ -707,16 +707,12 @@ func (m *ExpirationManager) LazyRevoke(ctx context.Context, leaseID string) erro
 	}
 
 	le.ExpireTime = time.Now()
-	{
-		m.pendingLock.Lock()
-		if err := m.persistEntry(ctx, le); err != nil {
-			m.pendingLock.Unlock()
-			return err
-		}
-
-		m.updatePendingInternal(le)
+	// TODO: potential race with concurrent renew.
```

Comment:
Should we be stopping the timer in this function to reduce concurrent renews?

Reply:
I didn't feel good about that.  The reasons were:
  * we're not doing a lot (no actual revoke call, etc.) so the scope is limited
  * on persistEntry error we have to re-insert
  
Now that I solved the second point for renew() -- sort of -- the same idea could be done here.

Reply:
Latest revision does call stopPending, and clones the old lease entry for re-insertion on storage failure.